src/Transactions/CreateTransaction/CreateTransactionContainer.js

```diff
@@ -23,32 +23,47 @@ import {
   transfersResource,
 } from '../../common/resources';
 import {
+  ALLOCATION_TYPE,
   TRANSACTION_SOURCE,
 } from '../constants';
 import CreateTransactionModal from './CreateTransactionModal';
 import { handleCreateTransactionErrorResponse } from './utils';
 
 export const CreateTransactionContainer = ({
+  allocationType,
   budgetName,
-  transactionType,
+  fetchBudgetResources,
+  fiscalYearCurrency,
   fiscalYearId,
   fundId,
+  labelId,
   mutator,
   onClose,
   stripes,
-  fetchBudgetResources,
-  intl,
-  fiscalYearCurrency,
+  transactionType,
 }) => {
+  const intl = useIntl();
   const showCallout = useShowCallout();
   const locale = stripes.locale;
   const currency = fiscalYearCurrency || stripes.currency;
 
   const transactionTypeKey = transactionType.toLowerCase();
-  const initialValues = useMemo(() => ({
-    fundId,
-    toFundId: fundId,
-  }), [fundId]);
+  const initialValues = useMemo(() => {
+    const values = allocationType === ALLOCATION_TYPE.decrease ? { fromFundId: fundId } : { toFundId: fundId };
+
+    return ({
+      fundId,
+      ...values,
+    });
+  }, [fundId, allocationType]);
+
+  const allocationProps = useMemo(() => {
```

Comment:
imo container shouldn't prepare such props, it's enough to drill `allocationType` prop to `CreateTransactionModal` and hide\show fields there